ui/dashboards/src/context/TemplateVariableProvider/TemplateVariableProvider.tsx

```diff
@@ -416,13 +416,43 @@ export interface TemplateVariableProviderProps {
    */
   externalVariableDefinitions?: ExternalVariableDefinition[];
   builtinVariables?: BuiltinVariableDefinition[];
+  enabledURLParams?: boolean;
 }
 
 export function TemplateVariableProvider({
   children,
   initialVariableDefinitions = [],
   externalVariableDefinitions = [],
   builtinVariables = [],
+  enabledURLParams = true,
```

Comment:
Enabled by default? 🤔 
IMO, we should provide minimum features and user enable whatever he wants

Reply:
Hi @jgbernalp What would you think of these breaking change from redhat? Are you using enabledURLParams ? Do you mind if we make the modifications that I listed in this previous message?

Reply:
Hi @celian-garcia, I think the correct way to go will be to decouple these providers. IMO the TimeRangeProvider should not rely on the URL state. That should be a choice when using the component, otherwise it comes to this issues in which the component is not reusable if you happen to have other means to store the range

Reply:
Yep absolutely I think we agree on that and that's exactly why I made this PR to make sure that this existing ``enabledURLParams`` boolean will actually ensure that the QueryParamsProvider is not necessary if flase.
Except if you'd have better way to decouple in head?

The breaking change would come with the changes proposed by @Gladorme.
- if we change the existing ``enabledURLParams`` to ``isURLParamsEnabled`` in TimeRangeProvider.. People using ``enabledURLParams`` will have to rename.
- if we make the new ``isURLParamsEnabled`` by default to false in TemplateVariableProvider (the current behaviour is that QueryParamProvider is needful by default). People using currently the TemplateVariableProvider  will have to explicitely set enabeldURLParams to true if they want the same behaviour

Reply:
A better way to decouple would be invert control and pass `timeRange` `setTimeRange` to the TimeRangeProvider, then you can choose to use them from the QueryParamsProvider or any other you choose. I think this boolean condition does not decouple the providers as consumers must know that using `false` will allow to use without the QueryProvider otherwise there will be an error.

Reply:
Ah yeah but then it becomes complex for the user who actually use the ``QueryParamsProvider`` they would have to run themselves the ``useQueryParams`` or use a wrapper that we would expose like ``useTimeRangeContextFromQueryParam`` and build from that the time range context (because there's not only time range but also refresh interval)

It would be fine to me but what I don't like is that it constraint to create a component that would be place under the QueryParamsProvider. Exit the straightforward way to integrate like in the doc https://github.com/perses/perses/blob/main/docs/embedded-panels.md#minimal-code
You may answer that nobody will anyway put all the providers like that in one only component, but this is exactly what I'm doing currently as I'm using a Perses panel in Angular environment :laughing: 

I'd be in favour of keeping what I did (maybe renaming the boolean to usingQueryParams) but with a default value to false. So that by default no error nothing specific apart from the given TimeRange. 
Note that if somebody want to modify the time range, they still can use the ``useTimeRange`` and modify from there. I even placed directly the ``<TimeRangeControls />`` in my code under the TimeRangeProvider and everything works well!

Reply:
Thank you for your participation I updated the PR with the code you gave to me.
I also adapted the `TemplateVariableProvider` to follow more or less the same principles except that I didn´t put the setter as it was a bit more complex than the time range provider (here there is zustand store with immer to partially modify variable definitions)

I will resolve this discussion as the breaking changes are properly listed in the PR description